### PR TITLE
PORTF-916 u boot add command for resetting ethernity

### PR DESCRIPTION
**Pull request by yba-siklu**
## Description
Added Ethernity cli reset command.

This PR also has a new version of cpld_reg.h for the EH8020 verified against "CPLD Register Description.xlsx".

## Test
 * None, still in debugging mode

## References
JIRA:PORTF-1082, PORTF-916

### CODE REVIEW

File: board/freescale/mx6ullevk/mx6ullevk_siklu_ethernity.c [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+/*
+ *
+ */
+static int do_siklu_ethernity_reset(cmd_tbl_t * cmdtp, int flag, int argc,
+		char * const argv[]) {
+
+	int rc = 0;
+	T_CPLD_LOGIC_WD_RW_REGS reset_reg;
+
+	reset_reg.s.cfg_fpga_8020_rst = 1;
```
Comment (baruchsiach) [inline board/freescale/mx6ullevk/mx6ullevk_siklu_ethernity.c:27]: What about all other bits in `uint8`. They are left uninitialized here. I guess the CPLD currently ignores these bits. But a future CPLD might behave strangely when random bits are set.

File: board/freescale/mx6ullevk/cpld_reg.h [hunk 2]
```diff
@@ -123,46 +144,47 @@ typedef union
...
 
-typedef union
-{
-	u8 uint8;
-	struct
-	{
-		u8 cfg_rst_n:1;		// When asserted to '0' resets the Ethernity switch.
-	} s;
-} T_CPLD_LOGIC_ETHERNITY_REGS;
-
```
Comment (baruchsiach) [inline board/freescale/mx6ullevk/cpld_reg.h]: This removes the definition added in the previous commit. I would be nicer to squash this change into the previous commit.

File: board/freescale/mx6ullevk/mx6ullevk_siklu_ethernity.c (hunk 1 above)
Comment (yba-siklu) [reply board/freescale/mx6ullevk/mx6ullevk_siklu_ethernity.c:27]: Fixed. Thanks.

File: board/freescale/mx6ullevk/cpld_reg.h (hunk 2 above)
Comment (yba-siklu) [reply board/freescale/mx6ullevk/cpld_reg.h]: Fixed, thanks.

File: board/freescale/mx6ullevk/mx6ullevk_siklu_ethernity.c [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+	int rc = 0;
+	T_CPLD_LOGIC_WD_RW_REGS reset_reg;
+
+	reset_reg.uint8 = siklu_cpld_read(R_CPLD_LOGIC_WD_RW);
+	reset_reg.s.cfg_fpga_8020_rst = 1;
+
+	rc = siklu_cpld_write(R_CPLD_LOGIC_WD_RW, reset_reg.uint8);
+	if (rc != 0) {
+		printf("%s() ERROR writing to CPLD, line %d\n", __func__, __LINE__);
+		return rc;
```
Comment (baruchsiach) [inline board/freescale/mx6ullevk/mx6ullevk_siklu_ethernity.c]: Nitpick: this line is redundant.

**Review by baruchsiach — DISMISSED** [review]

File: board/freescale/mx6ullevk/mx6ullevk_siklu_ethernity.c (hunk 3 above)
Comment (yba-siklu) [reply board/freescale/mx6ullevk/mx6ullevk_siklu_ethernity.c]: Fixed.

**Review by baruchsiach — APPROVED** [review]

**Pull request merged**